elasticsearch_test.go

```diff
@@ -47,105 +45,78 @@ func (e *UnitTestEnv) TestElasticsearch_Type(t *testing.T) {
+	statements1 := newdbplugin.Statements{
+		Commands: []string{`{"elasticsearch_role_definition": {"indices": [{"names":["*"], "privileges":["read"]}]}}`},
+	}
+	req1 := newdbplugin.NewUserRequest{
+		UsernameConfig: newdbplugin.UsernameMetadata{
+			DisplayName: "display-name",
+			RoleName:    "role-name",
+		},
+		Statements: statements1,
 	}
-	if username1 == "" {
+	resp1 := dbtesting.AssertNewUser(t, e.Elasticsearch, req1)
+	if resp1.Username == "" {
 		t.Fatal("expected username")
 	}
-	if password1 == "" {
-		t.Fatal("expected password")
-	}
-	e.TestUsers[username1] = statements1
+	e.TestUsers[resp1.Username] = statements1
 
-	statements2 := dbplugin.Statements{
-		Creation: []string{`{"elasticsearch_roles": ["vault"]}`},
+	statements2 := newdbplugin.Statements{
+		Commands: []string{`{"elasticsearch_roles": ["vault"]}`},
 	}
-	username2, password2, err := e.Elasticsearch.CreateUser(context.Background(), statements2, usernameConfig, time.Time{})
-	if err != nil {
-		t.Fatal(err)
+	req2 := newdbplugin.NewUserRequest{
+		UsernameConfig: newdbplugin.UsernameMetadata{
+			DisplayName: "display-name",
+			RoleName:    "role-name",
+		},
+		Statements: statements2,
 	}
-	if username2 == "" {
+	resp2 := dbtesting.AssertNewUser(t, e.Elasticsearch, req2)
+	if resp2.Username == "" {
 		t.Fatal("expected username")
 	}
-	if password2 == "" {
-		t.Fatal("expected password")
-	}
-	e.TestUsers[username2] = statements2
+	e.TestUsers[resp2.Username] = statements2
 }
 
-func (e *UnitTestEnv) TestElasticsearch_RevokeUser(t *testing.T) {
+func (e *UnitTestEnv) TestElasticsearch_DeleteUser(t *testing.T) {
 	for username, statements := range e.TestUsers {
-		if err := e.Elasticsearch.RevokeUser(context.Background(), statements, username); err != nil {
-			t.Fatal(err)
+		req := newdbplugin.DeleteUserRequest{
+			Username:   username,
+			Statements: statements,
 		}
+		dbtesting.AssertDeleteUser(t, e.Elasticsearch, req)
 	}
 }
 
-func (e *UnitTestEnv) TestElasticsearch_RotateRootCredentials(t *testing.T) {
-	originalConfig := map[string]interface{}{
-		"username": e.Username,
-		"password": e.Password,
-		"url":      e.URL,
-	}
-	configToStore, err := e.Elasticsearch.RotateRootCredentials(context.Background(), nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if len(originalConfig) != len(configToStore) {
-		t.Fatalf("expected %s, received %s", originalConfig, configToStore)
-	}
-	for k, v := range originalConfig {
-		if k == "password" {
-			if configToStore[k] == v {
-				t.Fatal("password should have changed")
-			}
-			continue
-		}
-		if configToStore[k] != v {
-			t.Fatalf("for %s, expected %s but received %s", k, v, configToStore[k])
-		}
+func (e *UnitTestEnv) TestElasticsearch_UpdateUser(t *testing.T) {
+	req := newdbplugin.UpdateUserRequest{
+		Username: e.Username,
+		Password: &newdbplugin.ChangePassword{
+			NewPassword: "new password",
+		},
 	}
+	dbtesting.AssertUpdateUser(t, e.Elasticsearch, req)
```

Comment:
Also assert that you can log in with the new credentials?

Reply:
That is already being done in the integration tests: https://github.com/hashicorp/vault-plugin-database-elasticsearch/blob/9cde79b1bc67ea87451c3d96f33cbd33b19f5e7f/integration_test.go#L152-L170

Seems like that's a better spot since these tests are using a mock elasticsearch.